devel/tapi/Portfile

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+
+name                tapi
+version             1000.10.8
+revision            0
+categories          devel
+platforms           darwin
+maintainers         {kencu @kencu} openmaintainer
+
+license             APSL-2
+description         ${name} adds ld64 linker support for text-based API libraries.
+long_description    ${description} Newer MacOS SDKs contain only these TAPI stubs.
+
+homepage            http://opensource.apple.com/source/${name}/
+master_sites        http://opensource.apple.com/tarballs/${name}:tapi
+
+distfiles           ${name}-${version}.tar.gz:tapi
+
+checksums           tapi-1000.10.8.tar.gz \
+                    rmd160  f5aad7f6ed579ce83d1206910ea8885043b4ebdb \
+                    sha256  827e996529974305ef7933f3fa790f7ed068caa29db8f8c30b8a83c6826503f7 \
+                    size    205606
+
+depends_build-append port:python27 port:libxml2
+configure.args-append -DPYTHON_EXECUTABLE=${prefix}/bin/python2.7
+
+worksrcdir          tapi_build
+post-extract {
+    file mkdir ${workpath}/${worksrcdir}
+    file rename ${workpath}/${name}-${version} ${worksrcpath}/tapi
+}
+
+compiler.cxx_standard 2011
+
+configure.pre_args-replace -DCMAKE_BUILD_TYPE=MacPorts \
+                           -DCMAKE_BUILD_TYPE=Release
+configure.post_args ${worksrcpath}/llvm
+
+variant tapiswift description "Build with swift infrastructure" {
+
+    # this is the official Apple TAPI build method, using Apple's llvm/clang fork
+    # it is also possible to build TAPI with llvm-7 and above
+    # these will be added later
+
+    master_sites-append https://github.com/apple/swift-llvm/archive:llvm  \
+                        https://github.com/apple/swift-clang/archive:clang
+
+    distfiles-append    swift-4.2.2-RELEASE.tar.gz:llvm \
+                        swift-4.2.4-RELEASE.tar.gz:clang
```

Comment:
As above, I think you can move the distfiles labelling directly into the `checksums-append` below.

Reply:
Up until this moment in time, I was unaware that base might infer a labelled distfile from checksum line...I will see if that works.

Reply:
likewise, just fails.

Reply:
OK. I don't have time to look into it.